### PR TITLE
Emma green/fix receipt page and send discount to paypal

### PR DESCRIPTION
**Pull request by emma-green**

### CODE REVIEW

File: ecommerce/extensions/payment/views/cybersource.py [hunk 1]
```diff
@@ -222,7 +222,7 @@ def _get_basket(self, basket_id):
                 lms_discount_client = EdxRestApiClient(discount_lms_url,
                                                        jwt=self.request.site.siteconfiguration.access_token)
                 ck = basket.lines.first().product.course_id
-                user_id = self.request.user.id
+                user_id = self.request.user.lms_user_id
                 try:
                     response = lms_discount_client.user(user_id).course(ck).get()
                     self.request.POST = self.request.POST.copy()
```
Comment (MatthewPiatetsky) [inline ecommerce/extensions/payment/views/cybersource.py:228]: what is this line doing?

File: ecommerce/static/js/pages/basket_page.js [hunk 2]
```diff
@@ -544,11 +544,16 @@ define([
                         deferred = new $.Deferred(),
                         promise = deferred.promise(),
                         paymentProcessor = $btn.data('processor-name'),
+                        discountJwt = $btn.closest('#paymentForm').find('input[name="discount_jwt"]'),
```
Comment (MatthewPiatetsky) [inline ecommerce/static/js/pages/basket_page.js:547]: why not just select the input directly?

Comment (MatthewPiatetsky) [inline ecommerce/static/js/pages/basket_page.js:547]: why not just select the input directly?

File: ecommerce/extensions/api/v2/views/checkout.py [hunk 3]
```diff
@@ -31,6 +31,9 @@ def post(self, request):
             'Checkout view called for basket [%s].',
             basket_id
         )
+        
+        request._request.POST = request._request.POST.copy()
```
Comment (MatthewPiatetsky) [inline ecommerce/extensions/api/v2/views/checkout.py]: what is this line doing?

**Review by MatthewPiatetsky — APPROVED** [review]
+1 assuming paypal now displays discount for regular ecommerce

File: ecommerce/extensions/payment/views/cybersource.py (hunk 1 above)
Comment (emma-green) [reply ecommerce/extensions/payment/views/cybersource.py:228]: the lms_user_id is different from the user.id. The user.id is a serialized id that originated in the user table in ecommerce. The lms_user_id is the same as the user id in the lms. We want to ask for the user with the correct corresponding id.

Comment (MatthewPiatetsky) [reply ecommerce/extensions/payment/views/cybersource.py:228]: it sounds like your answer is about line 225? i was asking about line 228

File: ecommerce/extensions/api/v2/views/checkout.py (hunk 3 above)
Comment (emma-green) [reply ecommerce/extensions/api/v2/views/checkout.py]: The request object is not editable, so in order to edit it, we need to copy it first.

File: ecommerce/extensions/payment/views/cybersource.py (hunk 1 above)
Comment (emma-green) [reply ecommerce/extensions/payment/views/cybersource.py:228]: Ah. That line is not part of the pr, but the request object is not editable, so in order to edit it, we need to copy it first.

File: ecommerce/static/js/pages/basket_page.js (hunk 2 above)
Comment (bbaker6225) [reply ecommerce/static/js/pages/basket_page.js:547]: +1 on this question

**Review by bbaker6225 — APPROVED** [review]
🤷‍♂

**Pull request merged**